cmd/fpga_plugin/fpga_plugin.go

```diff
@@ -78,28 +82,56 @@ func discoverFPGAs(sysfsDir string, devfsDir string) (map[string]map[string]devi
 	for _, fpgaFile := range fpgaFiles {
 		fname := fpgaFile.Name()
 		if deviceReg.MatchString(fname) {
+			var interfaceId string
+
 			deviceFolder := path.Join(sysfsDir, fname)
 			deviceFiles, err := ioutil.ReadDir(deviceFolder)
 			if err != nil {
 				return nil, err
 			}
-			afuNodes := make(map[string][]string)
+			fpgaNodes := make(map[string][]string)
+
+			if mode == "region" {
+				for _, deviceFile := range deviceFiles {
+					name := deviceFile.Name()
+					if fmeReg.MatchString(name) {
+						if len(interfaceId) > 0 {
+							return nil, fmt.Errorf("Detected more than one FPGA region for device %s. Only one region per FPGA device is supported", fname)
+						}
+						interfaceIdFile := path.Join(deviceFolder, name, "pr", "interface_id")
+						data, err := ioutil.ReadFile(interfaceIdFile)
+						if err != nil {
+							return nil, err
+						}
+						interfaceId = strings.TrimSpace(string(data))
+						fpgaNodes[interfaceId] = append(fpgaNodes[interfaceId], name)
+					}
+				}
+			}
+
 			for _, deviceFile := range deviceFiles {
 				name := deviceFile.Name()
 				if portReg.MatchString(name) {
-					afuFile := path.Join(deviceFolder, name, "afu_id")
-					data, err := ioutil.ReadFile(afuFile)
-					if err != nil {
-						return nil, err
+					if mode == "region" {
+						if len(interfaceId) == 0 {
+							return nil, fmt.Errorf("No FPGA region found for %s", fname)
+						}
+						fpgaNodes[interfaceId] = append(fpgaNodes[interfaceId], name)
+					} else {
```

Comment:
not good if we going to have more than two modes in the future. better explicitly check if it is about other supported mode.

Reply:
Added a default case to catch unsupported modes.